modules/openthread/platform/ble.c

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/types.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/drivers/gpio.h>
+#include <soc.h>
+#include <zephyr/sys/ring_buffer.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/gatt.h>
+
+/* Zephyr OpenThread integration Library */
+#include <zephyr/net/openthread.h>
+
+/* OpenThread BLE driver API */
+#include <openthread/platform/ble.h>
+
+/* Zephyr Logging */
+
+#define LOG_MODULE_NAME net_openthread_tcat
+#define LOG_LEVEL       CONFIG_OPENTHREAD_LOG_LEVEL
+
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define DEVICE_NAME     CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+
+#define MY_SERVICE_UUID 0xfffb
+#define RX_CHARACTERISTIC_UUID                                                                     \
+	BT_UUID_128_ENCODE(0x6bd10d8b, 0x85a7, 0x4e5a, 0xba2d, 0xc83558a5f220)
+#define TX_CHARACTERISTIC_UUID                                                                     \
+	BT_UUID_128_ENCODE(0x7fddf61f, 0x280a, 0x4773, 0xb448, 0xba1b8fe0dd69)
+
+#define BT_UUID_TCAT_SERVICE    BT_UUID_DECLARE_16(MY_SERVICE_UUID)
+#define BT_UUID_TCAT_SERVICE_RX BT_UUID_DECLARE_128(RX_CHARACTERISTIC_UUID)
+#define BT_UUID_TCAT_SERVICE_TX BT_UUID_DECLARE_128(TX_CHARACTERISTIC_UUID)
+
+#define PLAT_BLE_RING_BUF_SIZE     500
+#define PLAT_BLE_THREAD_STACK_SIZE 4200
+#define PLAT_BLE_THREAD_DEALY      500
+#define PLAT_BLE_MSG_DATA_MAX      CONFIG_BT_L2CAP_TX_MTU /* must match the maximum MTU size used */
+
+#define PLAT_BLE_MSG_CONNECT    0xFE
+#define PLAT_BLE_MSG_DISCONNECT 0xFF
+
+/* Zephyr Kernel Objects */
+
+static void otPlatBleThread(void *, void *, void *);
+uint8_t otPlatBleMsgBuf[PLAT_BLE_MSG_DATA_MAX];
+
+K_SEM_DEFINE(otPlatBleInitSemaphor, 0, 1);
+K_SEM_DEFINE(otPlatBleEventSemaphor, 0, 10000);
+RING_BUF_DECLARE(otPlatBleRingBuf, PLAT_BLE_RING_BUF_SIZE);
+K_THREAD_DEFINE(otPlatBleTid, PLAT_BLE_THREAD_STACK_SIZE, otPlatBleThread, NULL, NULL, NULL, 5, 0,
+		PLAT_BLE_THREAD_DEALY);
```

Comment:
All those variables could be `static`? And please use `snake_case` for Zephyr code whenever possible (I am aware there have to exceptions for platform functions, but it's not the case for those variables).

Reply:
fixed.